### ISSUE TITLE
Pin the GitHub actions we use on this repo to a full length commit SHA

### ISSUE THREAD

**Issue by HonkingGoose** [renovatebot/renovate#10841]
**What would you like Renovate to be able to do?**

@rarkins and @viceice now that PR #10835 is merged, we can start thinking about pinning our GitHub Actions to the current full length Git commit SHA?

**Did you already have any implementation ideas?**

@viceice can you explain what pattern we need to follow to make things work properly?
I think you can use a comment to say what "tagged version" you're at???

As an example I've grabbed the commit to which `actions/setup-node@v2.2.0` points right now.

```yml
- name: Set up Node.js ${{ env.NODE_VERSION }}
  uses: actions/setup-node@38d90ce44d5275ad62cc48384b3d8a58c500bb5f
  with:
    node-version: ${{ env.NODE_VERSION }}
    cache: yarn
```

Or maybe we can add a comment which says what tag we're "following".

```yml
- name: Set up Node.js ${{ env.NODE_VERSION }}
  uses: actions/setup-node@38d90ce44d5275ad62cc48384b3d8a58c500bb5f # setup-node: tag=v2
  with:
    node-version: ${{ env.NODE_VERSION }}
    cache: yarn
```

We'll also need to update the `default.json` over on the `renovate/.github` repository, to make use of the new feature.

**Comment by viceice** [comment]
1. We need to wait for `v25.56.0` to be live in hosted app
2.  We need to add `helper:pinGitHubActionDigests` preset to 
  - https://github.com/renovatebot/.github/blob/main/default.json
  - https://github.com/containerbase/.github/blob/main/default.json

**Comment by HonkingGoose** [comment]
I've made PRs on `renovatebot/.github` and on `containerbase/.github` to add `helper:pinGitHubActionDigests` to their config files.

See: 

- https://github.com/renovatebot/.github/pull/23
- https://github.com/containerbase/.github/pull/2

**Comment by rarkins** [comment]
@viceice how come we already got so many pinning PRs in this org already? Is it because we have pinDigests=true already?

**Comment by viceice** [comment]
i did not found any ref to `pinDigests=true`. i don't know what happens 🤔

**Comment by viceice** [comment]
on `docker-rust` we don't have pin updates
![image](https://user-images.githubusercontent.com/1798109/125760709-77deccf5-2f3d-41b3-b998-b5010107ce68.png)

on `internal-tools` we have pin updates
![image](https://user-images.githubusercontent.com/1798109/125760877-ecc9d542-a1e3-4edf-a82e-a8b2fe93f93e.png)

**Comment by rarkins** [comment]
Those PRs caused config errors so I have reverted them. I'd love to know what's going on..

**Comment by viceice** [comment]
> Those PRs caused config errors so I have reverted them. I'd love to know what's going on..

It was wrong preset name, there was a missing `s`, so right name is `helpers:pinGitHubActionDigests`, updated my comment above too.

But it's unclear to my. why we have pin pr's without even adding this preset. And why only in some repos. Very strange.

**Comment by HonkingGoose** [comment]
> It was wrong preset name, there was a missing `s`, so right name is `helpers:pinGitHubActionDigests`, updated my comment above too.

Opened PR #10857 to fix the typo in the docs.

**Comment by HonkingGoose** [comment]
> Those PRs caused config errors so I have reverted them. I'd love to know what's going on..

The config error was probably because I used the wrong name for the `helpers:` preset... 😉 

Do you want 2 new PRs from me with the correct config, or do you want to hold off for now?

**Comment by viceice** [comment]
we should hold of until we know why we already got pin pr's.

**Comment by viceice** [comment]
Found it, `github-actions` assign `language=docker`

https://github.com/renovatebot/renovate/blob/bcb359d67f2b2954ffe2cbc123cba9212a2a7505/lib/manager/github-actions/index.ts#L4-L4

so it is assigning the `docker:pinDigests` preset:
https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/config/presets/internal/docker.ts#L36-L41

here: 
https://github.com/renovatebot/renovate/blob/11694e92949171da339e54fe0d31b1f69532a6ad/lib/config/index.ts#L23-L26

**Comment by viceice** [comment]
@HonkingGoose you can now prepare those pr's.

**Comment by HonkingGoose** [comment]
> @HonkingGoose you can now prepare those pr's.

You've got new PR's! 😉

**Comment by HonkingGoose** [comment]
This issue can be closed now that we merged PR #10858 and have pinned our GitHub Actions to a full length commit SHA.
